src/data/set/intervals/group.lean

```diff
@@ -98,4 +101,104 @@ end
 
 end linear_ordered_add_comm_group
 
+/-! ### Lemmas about disjointness of translates of intervals -/
+section pairwise_disjoint
+
+section linear_ordered_add_comm_group
+
+variables [linear_ordered_add_comm_group α] (a b : α)
+
+lemma pairwise_disjoint_Ioc_add_zsmul :
+  pairwise (disjoint on (λ (n:ℤ), Ioc (a + n • b) (a + (n + 1) • b))) :=
+begin
+  -- The assumption `linear_ordered_add_comm_group α` can be weakened to `ordered_add_comm_group α`
+  -- if we assume that `0 < b`, but `Ioc` is not a very useful concept in partial orders anyway.
```

Comment:
I'm not sure I believe your explanation. You shouldn't need `b > 0` to generalise the statement, as `Ioc (a + n \smul b) (a + (n + 1) \smul b)` already implies `b > 0`. Your proof will need tweaking however.

Reply:
Proof tweaked accordingly.